src/registry/domain/components-details.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const async = require('async');
+const _ = require('lodash');
+
+const eventsHandler = require('./events-handler');
+const getUnixUTCTimestamp = require('../../utils/get-unix-utc-timestamp');
+
+module.exports = (conf, cdn) => {
+
+  const returnError = (code, message, callback) => {
+    eventsHandler.fire('error', { code, message });
+    return callback(code);
+  };
+
+  const filePath = () => `${conf.s3.componentsDir}/components-details.json`;
+
+  const getFromJson = callback => cdn.getJson(filePath(), true, callback);
+
+  const getFromDirectories = (options, callback) => {
+
+    const details = _.extend({}, _.cloneDeep(options.details));
```

Comment:
options.details can be empty in case it's the first time the new version is running with the current registry. So, options.details can be undefined and I need details to fallback to empty object in that scenario.